transport/wsn/src/test/java/org/kaazing/gateway/transport/wsn/specification/ws/ClosingHandshakeIT.java

```diff
@@ -0,0 +1,190 @@
+ * limitations under the License.
+ */
+
+package org.kaazing.gateway.transport.wsn.specification.ws;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.junit.rules.RuleChain.outerRule;
+
+import java.net.URI;
+
+import org.junit.Ignore;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.DisableOnDebug;
+import org.junit.rules.TestRule;
+import org.junit.rules.Timeout;
+import org.kaazing.gateway.server.test.GatewayRule;
+import org.kaazing.gateway.server.test.config.GatewayConfiguration;
+import org.kaazing.gateway.server.test.config.builder.GatewayConfigurationBuilder;
+import org.kaazing.k3po.junit.annotation.Specification;
+import org.kaazing.k3po.junit.rules.K3poRule;
+
+/**
+ * RFC-6455, section 7 "Closing the Connection"
+ */
+public class ClosingHandshakeIT {
+
+    private final K3poRule k3po = new K3poRule().setScriptRoot("org/kaazing/specification/ws/closing");
+    
+    private GatewayRule gateway = new GatewayRule() {
+        {
+            // @formatter:off
+            GatewayConfiguration configuration =
+                    new GatewayConfigurationBuilder()
+                        .service()
+                            .accept(URI.create("ws://localhost:8080/echo"))
+                            .type("echo")
+                            .crossOrigin()
+                                .allowOrigin("http://localhost:8001")
+                            .done()
+                        .done()
+                    .done();
+            // @formatter:on
+            init(configuration);
+        }
+    };
+
+    private final TestRule timeout = new DisableOnDebug(new Timeout(8, SECONDS));
+
+    @Rule
+    public final TestRule chain = outerRule(k3po).around(gateway).around(timeout);
+
+    @Test
+    @Specification({
+        "client.send.empty.close.frame/handshake.request.and.frame"})
+    public void shouldCompleteCloseHandshakeWhenClientSendEmptyCloseFrame() throws Exception {
+        k3po.finish();
+    }
+
+    @Test
+    @Specification({
+        "client.send.close.frame.with.code.1000/handshake.request.and.frame"})
+    public void shouldCompleteCloseHandshakeWhenClientSendCloseFrameWithCode1000() throws Exception {
+        k3po.finish();
+    }
+
+    @Test
+    @Specification({
+        "client.send.close.frame.with.code.1000.and.reason/handshake.request.and.frame"})
+    public void shouldCompleteCloseHandshakeWhenClientSendCloseFrameWithCode1000AndReason() throws Exception {
+        k3po.finish();
+    }
+
+    @Test
+    @Ignore("Read code different than expected and did not fail")
+    @Specification({
+        "client.send.close.frame.with.code.1000.and.invalid.utf8.reason/handshake.request.and.frame"})
+    public void shouldFailWebSocketConnectionWhenClientSendCloseFrameWithCode1000AndInvalidUTF8Reason() throws Exception {
+        k3po.finish();
+    }
+
+    @Test
+    @Ignore("Read code off by byte, not matching spec test")
```

Comment:
@jitsni 
Here expected is:
```
read [0x88 0x02 0x03 0xe8]

closed
```
but actual is 
```
read [0x88 0x02 0x03 0xe9]
```

Again, gateway is also not closing. Is this a script, gateway, or test login issue?